tinygrad/codegen/uops.py

```diff
@@ -460,8 +460,9 @@ def flops_mem(self, ignore_indexing=False) -> Tuple[sint, sint]:
   def type_verify(self):
     for u in self.uops:
       uop, arg, src, dtype = u.op, u.arg, u.src, u.dtype
-      if uop in {UOps.CONST, UOps.DEFINE_ACC}:
-        if uop is UOps.DEFINE_ACC: arg = arg[0]
+      if uop is UOps.DEFINE_ACC:
+        assert dtype is not None and src[0].dtype is dtype.scalar(), f"type of {src[0].dtype=} must be a scalar {dtype.scalar()}"
+      if uop is UOps.CONST:
         assert dtype is not None and type(arg) is type(dtypes.as_const(arg, dtype)), f"type of {arg=} does not match {dtype}"
```

Comment:
The dtypes.as_const assert should still work?

Reply:
Yes, changed.